src/lib/desmoHub-module.ts

```diff
@@ -322,4 +322,38 @@ export class DesmoHub {
     const { url, owner, disabled, score }: ITDD = await this.contract.getTDD();
     return { url, owner, disabled, score } as ITDD;
   }
+
+  /**
+   * @param start the initial TDD index to retrieve (by default is 0)
+   * @param end the final TDD index to retrieve (by default is the index of the last TDD in contract's storage)
+   * @returns an array containing the requested TDD descriptions.
+   */
+  public async getTDDList(start?: number, end?: number): Promise<ITDD[]> {
+    const storageLength = (await this.getTDDStorageLength()).toNumber();
+
+    if (storageLength < 1) {
+      throw new Error('No TDDs are currently registered in the contract.');
+    }
+
+    // We must be sure that `start` and `end` are integer numbers:
+    start = Math.floor(start ?? 0);
+    end = Math.floor(end ?? storageLength - 1);
+
+    if (start > end) {
+      throw new Error(`Start index (${start}) greater than end index (${end}).`);
+    }
+
+    const promises: Promise<ITDD>[] = [];
+    for (let i = start; i < end + 1; ++i) {
+      promises.push(this.contract.getTDDByIndex(i));
+    }
+    const tddList = await Promise.all(promises);
+
+    for (let i = 0; i < tddList.length; ++i) {
+      const { url, owner, disabled, score }: ITDD = tddList[i];
+      tddList[i] = { url, owner, disabled, score } as ITDD;
+    }

```

Comment:
Why are u simply casting each element of the list to ITDD?

Reply:
The objects that we get from the contract function are of a strange type.
The `url`, `owner`, `disabled` and `score` values are accessible both using a numerical index and using the property name.
For example, the `owner` can be obtained both as `tddObject[1]` and as `tddObject.owner`.

Example:
```typescript
[
    'https://desmold-zion-2.vaimee.it',
    '0x70997970C51812dc3A010C7d01b50e0d17dc79C8',
    false,
    BigNumber { _hex: '0x00', _isBigNumber: true },
    url: 'https://desmold-zion-2.vaimee.it',
    owner: '0x70997970C51812dc3A010C7d01b50e0d17dc79C8',
    disabled: false,
    score: BigNumber { _hex: '0x00', _isBigNumber: true }
  ]
```

What I'm doing in those lines of code is to clean up those objects from the numeric properties.
**I'm doing the same in the `getTDD()` method.**